doc/releases/changelog-0.30.0.md

```diff
@@ -149,53 +288,115 @@
+  
+* `pennylane.devices.qubit.preprocess` now allows circuits with non-commuting observables.
+  [(#3857)](https://github.com/PennyLaneAI/pennylane/pull/3857)
+
+* `qml.devices.qubit.measure` now computes the expectation values of `Hamiltonian` and `Sum`
+  in a backpropagation-compatible way.
+  [(#3862)](https://github.com/PennyLaneAI/pennylane/pull/3862/)
+
+<h4>Pulse programming</h4>
+
+* Here are the functions, classes, and more that were added or improved to facilitate simulating ensembles of Rydberg atoms:
+  [(#3749)](https://github.com/PennyLaneAI/pennylane/pull/3749)
+  [(#3911)](https://github.com/PennyLaneAI/pennylane/pull/3911)
+  [(#3930)](https://github.com/PennyLaneAI/pennylane/pull/3930)
+  [(#3936)](https://github.com/PennyLaneAI/pennylane/pull/3936)
+  [(#3966)](https://github.com/PennyLaneAI/pennylane/pull/3966)
+  [(#3987)](https://github.com/PennyLaneAI/pennylane/pull/3987)
+  [(#3889)](https://github.com/PennyLaneAI/pennylane/pull/3889)
+  [(#4021)](https://github.com/PennyLaneAI/pennylane/pull/4021)
+  + `HardwareHamiltonian`: an internal class that contains additional information about pulses and settings.
+  + `rydberg_interaction`: a user-facing function that returns a `HardwareHamiltonian` containing
+    the Hamiltonian of the interaction of all the Rydberg atoms.
+  + `transmon_interaction`: a user-facing function for constructing
+    the Hamiltonian that describes the circuit QED interaction Hamiltonian of superconducting transmon systems.
+  + `drive`: a user-facing function function that returns a `ParametrizedHamiltonian` (`HardwareHamiltonian`) containing
+    the Hamiltonian of the interaction between a driving electro-magnetic field and a group of qubits.
+  + `rydberg_drive`: a user-facing function that returns a `ParametrizedHamiltonian` (`HardwareHamiltonian`) containing
+    the Hamiltonian of the interaction between a driving laser field and a group of Rydberg atoms.
+  + `max_distance`: a keyword argument added to `qml.pulse.rydberg_interaction` to allow for the removal of negligible contributions from atoms beyond `max_distance` from each other.
+
+* `ParametrizedEvolution` now takes two new Boolean keyword arguments: `return_intermediate` and
+  `complementary`. They allow computing intermediate time evolution matrices.
+  [(#3900)](https://github.com/PennyLaneAI/pennylane/pull/3900)
+  
+  Activating `return_intermediate` will return intermediate time evolution steps, for example
+  for the matrix of the Operation, or of a quantum circuit when used in a QNode.
+  Activating `complementary` will make these intermediate steps be the _remaining_
+  time evolution complementary to the output for `complementary=False`.
+  See the [docstring](https://docs.pennylane.ai/en/stable/code/api/pennylane.pulse.ParametrizedEvolution.html)
+  for details.
+
+* `ParametrizedEvolution` takes two new Boolean keyword arguments: `return_intermediate` and
```

Comment:
Duplicate entries here but slightly different content in the descriptions 🤔. Which one is best to go off of?